providers/github/connection/connection.go

```diff
@@ -67,8 +68,10 @@ func NewGithubConnection(id uint32, asset *inventory.Asset) (*GithubConnection,
 		}
 	}
 
+	ctx := context.WithValue(context.Background(), github.SleepUntilPrimaryRateLimitResetWhenRateLimited, true)
```

Comment:
Can we add a comment with your research findings, so that this context is written down in code?

Reply:
Added the comment